### PR TITLE
add organization and dns to container variables

### PR DESCRIPTION
**Pull request by ashleyschuett**
@normanjoyner

### CODE REVIEW

File: lib/follower.js [hunk 1]
```diff
@@ -75,8 +76,14 @@ module.exports = {
...
 
+            const dns_hash = crypto.md5(`${config.organization}.${core.options.cluster_id}`);
+            const dns_entry = `${application_name}.${dns_hash}.dns.cship.co`;
+            const app_url = `http://${dns_entry}`;
+            container.env_vars[`CS_DNS_ADDRESS_${application_name.toUpperCase()}`] = app_url;
+            container.env_vars.CS_ORGANIZATION_ID = config.organization;
+
             let options = {
-                url: `${constants.CLOUD_API_BASE_URL}/v2/organizations/${config.organization}/clusters/${core.cluster_id}`,
+                url: `${constants.CLOUD_API_BASE_URL}/v2/organizations/${config.organization}/clusters/${core.options.cluster_id}`,
```
Comment (normanjoyner) [inline lib/follower.js]: we need to use `core.cluster_id` here in the case that a user did not specify `--cluster-id` when launching the containership agent

File: lib/follower.js [hunk 2]
```diff
@@ -75,8 +76,14 @@ module.exports = {
             let application_name = container_options.application_name;
             let container = _.omit(container_options, ['application_name', 'start_args']);
 
+            const dns_hash = crypto.md5(`${config.organization}.${core.options.cluster_id}`);
```
Comment (normanjoyner) [inline lib/follower.js]: `core.cluster_id`

File: lib/follower.js [hunk 3]
```diff
@@ -75,8 +76,14 @@ module.exports = {
...
+            const dns_entry = `${application_name}.${dns_hash}.dns.cship.co`;
+            const app_url = `http://${dns_entry}`;
+            container.env_vars[`CS_DNS_ADDRESS_${application_name.toUpperCase()}`] = app_url;
```
Comment (normanjoyner) [inline lib/follower.js]: change environment variable to `CS_CLOUD_DNS_ADDRESS...` to match containership specific environment variables prefixed with `CS_`

File: lib/follower.js [hunk 4]
```diff
@@ -75,8 +76,14 @@ module.exports = {
...
+            container.env_vars.CS_ORGANIZATION_ID = config.organization;
```
Comment (normanjoyner) [inline lib/follower.js]: change environment variable to `CS_CLOUD_DNS_ADDRESS...` to match containership specific environment variables prefixed with `CS_`

File: lib/follower.js [hunk 5]
```diff
@@ -75,8 +76,14 @@ module.exports = {
             let application_name = container_options.application_name;
             let container = _.omit(container_options, ['application_name', 'start_args']);
 
+            const dns_hash = crypto.md5(`${config.organization}.${core.options.cluster_id}`);
+            const dns_entry = `${application_name}.${dns_hash}.dns.cship.co`;
+            const app_url = `http://${dns_entry}`;
```
Comment (normanjoyner) [inline lib/follower.js]: can remove this

File: lib/follower.js (hunk 3 above)
Comment (normanjoyner) [inline lib/follower.js]: rather than `app_url`, can just use:
```javascript
`http://${dns_entry}`
```

File: lib/follower.js (hunk 2 above)
Comment (ashleyschuett) [inline lib/follower.js]: core.cluster_id is always undefined

Comment (normanjoyner) [inline lib/follower.js]: `core.cluster_id` *should* be available, so we need to dig into why this is coming back `undefined`.

Comment (ashleyschuett) [inline lib/follower.js]: @normanjoyner @nicktate 
Cluster id get's set in myraid here 
https://github.com/containership/containership.core/blob/c1274ca9683b9e299d29b5255e96a153430a62e3/lib/applications.js#L24

which can only ever be called if the server is in leader mode 
https://github.com/containership/containership.core/blob/master/containership-core.js#L30

Should the logic for setting the cluster id be moved to cluster instead?

**Pull request merged**